storage/s3.go

```diff
@@ -678,9 +684,29 @@ func newSession(opts Options) (*session.Session, error) {
 		sess.Config.Region = aws.String(endpoints.UsEast1RegionID)
 	}
 
+	// get region of the bucket and create session accordingly
+	// if the region is not provided, it means we want region-independent session
+	// for operations such as listing buckets, making a new bucket, ...
+	if opts.bucket != "" {
```

Comment:
As far as I understand from the code, we always fetch bucket region of all given distinct buckets. Am I right?

It should be enough to make a get-bucket-region request **if** it's a transfer request (cp/mv s3->s3) **and** given urls have different bucket names.

Reply:
thanks for the feedback. yes, we fetch bucket region for each distinct bucket. 

If we make get-bucket-region request only for transfer operations (cp/mv, as you suggested), we will still face an issue for batch operations. For instance,

`s5cmd ls | awk '{printf "ls %s/\n", $3}' | s5cmd run` will throw an error if all buckets reside on different regions.

Reply:
You're handling `ls` requests for region enabled S3 buckets too. My assumption of the change was based on the PR title (cross region transfer).

Reply:
oh, ok, thanks :)